### ISSUE TITLE
Sub-process invocations of JRuby with jruby-complete fail in 1.7.21

### ISSUE THREAD

**Issue by rtyler** [jruby/jruby#3141]
Here's [a gist with some example output between 1.7.20.1 and 1.7.21](https://gist.github.com/rtyler/aa4b08946b4e01edc777) when installing a gem that has a native extension component to it (e.g. `thrift` 0.9.1)

This is pretty easy to reproduce in [jruby-gradle](/jruby-gradle) land with the following build.gradle:

```gradle
buildscript {
    repositories { jcenter() }

    dependencies {
        classpath "com.github.jruby-gradle:jruby-gradle-plugin:0.4.0"
    }
}

apply plugin: 'com.github.jruby-gradle.base'

jruby {
    defaultVersion = jrubyVersion
    execVersion = jrubyVersion
}

dependencies {
    gems 'rubygems:thrift:0.9.1'
}
```

Then just invoke:

* *Failure:* `./gradlew -PjrubyVersion=1.7.21 jrubyPrepare`
* *Success:* `./gradlew -PjrubyVersion=1.7.20.1 jrubyPrepare`

Looks like something went awry in 1.7.21

**Comment by mkristian** [comment]
simpler way to reproduce
```
$ GEM_HOME=target java -jar maven/jruby-complete/target/jruby-complete-1.7.22-SNAPSHOT.jar -S gem install thrift

Building native extensions.  This could take a while...
ERROR:  Error installing thrift:
	ERROR: Failed to build gem native extension.

    "java -cp :/home/christian/projects/active/jruby/jruby17/maven/jruby-complete/target/jruby-complete-1.7.22-SNAPSHOT.jar org.jruby.Main" -r ./siteconf20150716-27293-r7lex4.rb extconf.rb
/bin/sh: 1: java -cp :/home/christian/projects/active/jruby/jruby17/maven/jruby-complete/target/jruby-complete-1.7.22-SNAPSHOT.jar org.jruby.Main: not found

extconf failed, uncaught signal 127

Gem files will remain installed in /home/christian/projects/active/jruby/jruby17/target/gems/thrift-0.9.2.0 for inspection.
Results logged to /home/christian/projects/active/jruby/jruby17/target/extensions/universal-java-1.7/1.9/thrift-0.9.2.0/gem_make.out
```

**Comment by mkristian** [comment]
fixed via 917d0e7ddad7ddbc792a274ebed3b933b3b6453f
